File: toolloop-front/src/app/core/models/dto/dashboard-info.ts
import {Tool} from '../entity/tool';
import {Rental} from '../entity/rental';

export interface DashboardInfo {
    totalEarnings: number;
    totalRentals: number;
    totalTools: number;
    activeRentals: number;
    userRating: number;
    nextExpiringRental: Rental | null;
    recentTools: Tool[];
}

File: toolloop-back/src/main/java/com/toolloop/model/dto/DashboardInfo.java
package com.toolloop.model.dto;

import com.toolloop.model.entity.Rental;
import com.toolloop.model.entity.Tool;
import io.quarkus.runtime.annotations.RegisterForReflection;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@Builder
@RegisterForReflection
@Data
public class DashboardInfo {
    public BigDecimal totalEarnings;
    public Integer totalRentals;
    public Integer totalTools;
    public Integer activeRentals;
    public BigDecimal userRating;
    public Rental nextExpiringRental;
    public List<Tool> recentTools;
}
